### PR TITLE
Created new Bridge method: getQueuedPegoutsCount

### PR DESCRIPTION
**Pull request by kelvinator07**

### CODE REVIEW

**Comment by kelvinator07** [comment]
pipeline:run

**Review by josedahlquist — APPROVED** [review]

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java [hunk 1]
```diff
@@ -7577,6 +7577,35 @@ public void getNextPegoutCreationBlockNumber_after_RSKIP271_activation() {
...
+        when(provider.getReleaseRequestQueue()).thenReturn(new ReleaseRequestQueue(Collections.emptyList()));
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+                .withProvider(provider)
+                .withActivations(activations)
+                .build();
+
+        assertEquals(0, bridgeSupport.getQueuedPegoutsCount());
+    }
+
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java:7608]: Let's add a test where release request queue is not empty

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java [hunk 2]
```diff
@@ -2504,6 +2504,17 @@ public long getNextPegoutCreationBlockNumber() {
         return activations.isActive(RSKIP271) ? provider.getNextPegoutHeight().orElse(0L) : 0L;
     }
 
+    public int getQueuedPegoutsCount() {
+        if (activations.isActive(RSKIP271)) {
+            try {
+                return provider.getReleaseRequestQueue().getEntries().size();
```
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java:2510]: Maybe we could a method in BridgeStorageProvider to return the release request queue size. What do you think? A bit nitpicky maybe, but I believe we are violating Demeter's law here

**Comment by kelvinator07** [comment]
pipeline: run

File: rskj-core/src/main/java/co/rsk/peg/BridgeMethods.java [hunk 3]
```diff
@@ -668,6 +668,17 @@
             (BridgeMethodExecutorTyped) Bridge::getNextPegoutCreationBlockNumber,
             activations -> activations.isActive(RSKIP271),
             fixedPermission(false)
+    ),
+    GET_QUEUED_PEGOUTS_COUNT(
+            CallTransaction.Function.fromSignature(
+                    "getQueuedPegoutsCount",
+                    new String[]{},
+                    new String[]{"int256"}
```
[the remark this answers is not in the document]
Comment (kelvinator07) [reply rskj-core/src/main/java/co/rsk/peg/BridgeMethods.java:676]: int256 or uint256?

[the remark this answers is not in the document]
Comment (marcos-iov) [reply rskj-core/src/main/java/co/rsk/peg/BridgeMethods.java:676]: uint, we shouldn't have a negative amount of pegout requests

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java (hunk 2 above)
Comment (kelvinator07) [reply rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java]: Done

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java [hunk 4]
```diff
@@ -7577,6 +7577,35 @@ public void getNextPegoutCreationBlockNumber_after_RSKIP271_activation() {
...
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP271)).thenReturn(false);
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+                .withActivations(activations)
+                .build();
+
+        assertEquals(0, bridgeSupport.getQueuedPegoutsCount());
+    }
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java:7591]: Let's add a mock for BridgeStorageProvider when creating BridgeSupport. Then verify `provider..getReleaseRequestQueueSize()` was never called.

Just to be sure

File: rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java [hunk 5]
```diff
@@ -912,6 +912,13 @@ protected void saveNextPegoutHeight() {
         safeSaveToRepository(NEXT_PEGOUT_HEIGHT_KEY, nextPegoutHeight, BridgeSerializationUtils::serializeLong);
     }
 
+    protected int getReleaseRequestQueueSize() {
+        if (releaseRequestQueue != null && activations.isActive(RSKIP271)) {
```
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java:916]: No need to check for activations here. The queue size is not a consensus change

Comment (marcos-iov) [reply rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java:916]: Also, you should be calling `getReleaseRequestQueue` method to get the queue from storage if not present in the memory

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java [hunk 6]
```diff
@@ -2504,6 +2504,13 @@ public long getNextPegoutCreationBlockNumber() {
         return activations.isActive(RSKIP271) ? provider.getNextPegoutHeight().orElse(0L) : 0L;
     }
 
+    public int getQueuedPegoutsCount() {
+        if (activations.isActive(RSKIP271)) {
+            return provider.getReleaseRequestQueueSize();
+        }
+        return 0;
```
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java:2511]:
```suggestion
        return activations.isActive(RSKIP271) ? provider.getReleaseRequestQueueSize() : 0;
```
Just a suggestion, if you like it it's shorter

**Review by marcos-iov — CHANGES_REQUESTED** [review]

File: rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java (hunk 5 above)
Comment (kelvinator07) [reply rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java]: Resolved

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java (hunk 6 above)
Comment (kelvinator07) [reply rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java]: Resolved

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java (hunk 4 above)
Comment (kelvinator07) [reply rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java:7590]: Resolved

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java [hunk 7]
```diff
@@ -2504,6 +2504,15 @@ public long getNextPegoutCreationBlockNumber() {
...
     }
 
+    public int getQueuedPegoutsCount() {
+        try {
+            return provider.getReleaseRequestQueueSize();
+        } catch (IOException e) {
+            logger.error("Unexpected error accessing storage while attempting to get the ReleaseRequestQueueSize", e);
+        }
+        return 0;
+    }
```
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java:2514]: In the Bridge we should check for RSKIP activation like you had before. Since this method did not exist prior to this HF.

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java (hunk 6 above)
Comment (marcos-iov) [reply rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java]: You removed RSKIP check

**Review by marcos-iov — CHANGES_REQUESTED** [review]
Left some comments

File: rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java [hunk 8]
```diff
@@ -912,6 +912,15 @@ protected void saveNextPegoutHeight() {
         safeSaveToRepository(NEXT_PEGOUT_HEIGHT_KEY, nextPegoutHeight, BridgeSerializationUtils::serializeLong);
     }
 
+    protected int getReleaseRequestQueueSize() throws IOException {
+        if (releaseRequestQueue != null) {
+            return releaseRequestQueue.getEntries().size();
+        }
```
[the remark this answers is not in the document]
Comment (kelvinator07) [reply rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java]: Resolved

File: rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java [hunk 9]
```diff
@@ -912,6 +912,15 @@ protected void saveNextPegoutHeight() {
...
+
+        releaseRequestQueue = getReleaseRequestQueue();
+        return releaseRequestQueue.getEntries().size();
```
[the remark this answers is not in the document]
Comment (kelvinator07) [reply rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java]: Done

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java (hunk 7 above)
Comment (kelvinator07) [reply rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java:2516]: Resolved

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java (hunk 6 above)
Comment (kelvinator07) [reply rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java]: Resolved

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java [hunk 10]
```diff
@@ -2504,6 +2504,17 @@ public long getNextPegoutCreationBlockNumber() {
         return activations.isActive(RSKIP271) ? provider.getNextPegoutHeight().orElse(0L) : 0L;
     }
 
+    public int getQueuedPegoutsCount() {
+        if (activations.isActive(RSKIP271)) {
+            try {
+                return provider.getReleaseRequestQueueSize();
+            } catch (IOException e) {
+                logger.error("Unexpected error accessing storage while attempting to get the ReleaseRequestQueueSize", e);
+            }
```
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java:2513]: I'm not sure it's a good idea to catch the exception here, if there's an error reading from the storage the function will return `0` which could be misleading. I'd suggest removing the try/catch block and letting the Bridge `execute` method handle it

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java (hunk 4 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java:7606]: I meant to add this in the pre RSKIP activation test.

`BridgeSupport bridgeSupport = bridgeSupportBuilder.withActivations(activations).withProvider(...)`

And then `verify(times(0), provider..getReleaseRequestQueueSize())`

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java (hunk 10 above)
Comment (kelvinator07) [reply rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java]: Resolved

**Review by marcos-iov — APPROVED** [review]
